### ISSUE TITLE
Access violation at address xxx in module libcrypto-1_1.dll

### ISSUE THREAD

**Issue by dimmaq** [libssh2/libssh2#295]
Continue https://github.com/libssh2/libssh2/issues/292
Build new code. Now get **Access violation at address 5044917F in module 'libcrypto-1_1.dll'. Write of address CDCDCDD5.** in  https://github.com/libssh2/libssh2/blob/b45333b2b6fe2e207176d9d93fb1f3966a4e68b9/src/hostkey.c#L960

```
-		keyctx	0x015b5c00 {public_key=0x015b07c0 {...} private_key=0xcdcdcdcd {...} }	libssh2_curve25519_keys *
		public_key	0x015b07c0 {...}	evp_pkey_st *
		private_key	0xcdcdcdcd {...}	evp_pkey_st *

```
address CDCDCDD5 
private_key	0xcdcdcdcd
?
Openssl from https://slproweb.com/products/Win32OpenSSL.html Win32 OpenSSL v1.1.1
Builder Visual Studio 2017

**Comment by dimmaq** [comment]
Do `hostkey_method_ssh_ed25519_init()` initialize *private_key*?
`hostkey_method_ssh_ed25519_init()` return 
```
-		ctx	0x00e38280 {public_key=0x00e33a30 {...} private_key=0xcdcdcdcd {...} }	libssh2_curve25519_keys *
		public_key	0x00e33a30 {...}	evp_pkey_st *
		private_key	0xcdcdcdcd {...}	evp_pkey_st *

```
**private_key	0xcdcdcdcd**

**Comment by willco007** [comment]
My fix needed to be calloc instead of malloc, fix incoming.

**Comment by willco007** [comment]
Fixed with 0974481.
